### ISSUE TITLE
Feature: Emulator Workflow for Inference

### ISSUE THREAD

**Issue by SBeairsto** [climagination/ClimatExML#34]
### Summary

This issue is to initiate the development of a clean, modular **Emulator** workflow to support inference over the test set using trained super-resolution models (CNN, WGAN, etc.). The emulator will handle model loading, input formatting, optional stochastic sampling, and batch inference.

### Goals

- Create a `BaseEmulator` class to standardize the interface
- Implement `HRStreamEmulator` to wrap the `HRStreamGenerator` model
- Support loading from `.ckpt` files with stored hyperparameters
- Support stochastic sampling (multiple realizations per input)
- Save outputs in `.zarr` format for post-processing

### Longer-Term Goals

- Extend to support diffusion-based, and/or temporal emulators
- Enable direct inference from cloud-hosted `.zarr` data
- Optionally include un-normalization as part of the output interface
